File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/GegoennterKundenstamm/GegoennteKunden/DTOs/IGegoennterKundeDetail.cs
using Contract.Architecture.Backend.Core.Contract.Logic.Modules.GegoenntesBankwesen.GegoennteBanken;
using System;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.GegoennterKundenstamm.GegoennteKunden
{
    public interface IGegoennterKundeDetail
    {
        Guid Id { get; set; }

        string Name { get; set; }

        string GegoennterName { get; set; }

        bool? GegoennterBoolean { get; set; }

        DateTime? GegoennterDateTime { get; set; }

        double? GegoennterDouble { get; set; }

        Guid? GegoennterGuid { get; set; }

        int? GegoennterInteger { get; set; }

        IGegoennteBank BesteBank { get; set; }
    }
}

File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/gegoennter-kundenstamm/gegoennte-kunden/dtos/i-gegoennter-kunde-detail.ts
import { GegoennteBank, IGegoennteBank } from 'src/app/model/gegoenntes-bankwesen/gegoennte-banken/dtos/i-gegoennte-bank';
import { ApiGegoennterKundeDetail } from './api/api-gegoennter-kunde-detail';

export interface IGegoennterKundeDetail {
    id: string;
    name: string;
    gegoennterName: string;
    gegoennterBoolean?: boolean;
    gegoennterDateTime?: Date;
    gegoennterDouble?: number;
    gegoennterGuid?: string;
    gegoennterInteger?: number;
    besteBank: IGegoennteBank;
}

export abstract class GegoennterKundeDetail {
    public static fromApiGegoennterKundeDetail(apiGegoennterKundeDetail: ApiGegoennterKundeDetail): IGegoennterKundeDetail {
        return {
            id: apiGegoennterKundeDetail.id,
            name: apiGegoennterKundeDetail.name,
            gegoennterName: apiGegoennterKundeDetail.gegoennterName,
            gegoennterBoolean: apiGegoennterKundeDetail.gegoennterBoolean,
            gegoennterDateTime: apiGegoennterKundeDetail.gegoennterDateTime,
            gegoennterDouble: apiGegoennterKundeDetail.gegoennterDouble,
            gegoennterGuid: apiGegoennterKundeDetail.gegoennterGuid,
            gegoennterInteger: apiGegoennterKundeDetail.gegoennterInteger,
            besteBank: GegoennteBank.fromApiGegoennteBank(apiGegoennterKundeDetail.besteBank),
        };
    }
}
